platform/security/core/security-core-services/src/main/java/ddf/security/http/impl/HttpSessionFactory.java

```diff
@@ -39,6 +43,37 @@
    */
   @Override
   public synchronized HttpSession getOrCreateSession(HttpServletRequest httpRequest) {
+    /*
+     * DDF-6587 - This check is made so that when simultaneous requests are received for the same
+     * context, the second request will not attempt to create a new session.  This method is
+     * synchronized so the first request will create a new session object and the second one
+     * look up the previously created session object from the jetty session cache.  This is
+     * necessary because at this point in the processing chain, jetty was not able to associate
+     * a session for either of the simultaneous requests.  Therefore, we need to manually attach
+     * the session to the second request.  Otherwise, jetty will attempt to recreate the same
+     * session object and fail, causing the second request to fail.
+     */
+    if (httpRequest instanceof Request) {
+      try {
+        Request request = (Request) httpRequest;
+        SessionHandler sessionHandler = request.getSessionHandler();
+        SessionCache sessionCache = sessionHandler.getSessionCache();
+        String sessionId =
+            sessionHandler.getSessionIdManager().newSessionId(request, System.currentTimeMillis());
```

Comment:
❓ Should we do null checking before trying to use the session handle and session cache?

Jetty often uses null values as default values or when recycling objects.